siwe.go

```diff
@@ -0,0 +1,462 @@
+package siwe
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/dchest/uniuri"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/relvacode/iso8601"
+)
+
+type MalformedMessage struct{ string }
+type ParsingFailed struct{ string }
+type ExpiredMessage struct{ string }
+type InvalidMessage struct{ string }
+type InvalidSignature struct{ string }
+
+func (m *ParsingFailed) Error() string {
+	return "Expired Message"
```

Comment:
Wrong message. Also what's the difference between `ParsingFailed` and `InvalidMessage`?

Reply:
I was using it to describe the regex parsing error, but it doesn't make much sense, everywhere else in field validation I was using InvalidMessage, so I'm gonna keep just that one.